src/routes/api/assets/patchById.js

```diff
@@ -1,16 +1,85 @@
 const logger = require("../../../logger");
-const { createSuccessResponse } = require("../../../response");
+const prisma = require("../../../model/data/prismaClient");
+const {
+  createSuccessResponse,
+  createErrorResponse,
+} = require("../../../response");
 
 /**
- * Update an asset by it's id
+ * Update an asset by its id
  */
-
-// eslint-disable-next-line no-unused-vars
 module.exports = async (req, res, next) => {
-  logger.debug(
-    { user: req.user, id: req.params.id },
-    `received request: PATCH /v1/assets/:assetId`,
-  );
+  try {
+    const assetId = Number(req.params.assetId);
+
+    // Add debug logging
+    logger.debug(
+      {
+        params: req.params,
+        assetId: assetId,
+        isNaN: isNaN(assetId),
+        user: req.user,
+        body: req.body,
+      },
+      "PATCH request details",
+    );
+
+    // Validate assetId is a number
+    if (isNaN(assetId)) {
+      return res.status(400).json(createErrorResponse(400, "invalid asset id"));
+    }
+
+    logger.debug(
+      { user: req.user, assetId, body: req.body },
+      "received request: PATCH /v1/assets/:assetId",
+    );
+
+    // Find the asset and its creator
+    const asset = await prisma.asset.findUnique({
+      where: { id: assetId },
+      include: {
+        user: {
+          select: {
+            hashedEmail: true,
+          },
+        },
+      },
+    });
+
+    // Add more debug logging
+    logger.debug(
+      {
+        assetFound: !!asset,
+        assetDetails: asset,
+        requestUser: req.user,
+      },
+      "Asset lookup results",
+    );
+
+    // Check if asset exists
+    if (!asset) {
+      return res.status(404).json(createErrorResponse(404, "asset not found"));
```

Comment:
suggestion: We can use `next` instead of `createErrorResponse` to reduce code duplication and improve readability. You can take a look at POST /assets to see how to do this.